ui/imports/shared/popups/ImportCommunityPopup.qml

```diff
@@ -1,82 +1,144 @@
         }
 
-        StyledTextArea {
+        StatusBaseText {
+            id: inputLabel
+            text: qsTr("Community key")
+            color: Theme.palette.directColor1
+            font.pixelSize: 15
+        }
+
+        StatusTextArea {
             id: keyInput
-            label: qsTr("Community private key")
             placeholderText: "0x0..."
-            customHeight: 110
-            anchors.top: infoText1.bottom
-            anchors.topMargin: Style.current.bigPadding
-            anchors.left: parent.left
-            anchors.right: parent.right
-
-            onTextChanged: {
-                importButton.enabled = root.validate(keyInput.text)
+            height: 110
+            Layout.fillWidth: true
+            // onTextChanged: {
+            //     importButton.enabled = root.validate(keyInput.text)
+            // }
+          }
+
+        StatusBaseText {
+            id: detectionLabel
+            Layout.fillWidth: true
+            horizontalAlignment: Text.AlignRight
+            verticalAlignment: Text.AlignVCenter
+            font.pixelSize: 13
+            visible: keyInput.text.trim() !== ""
+            text: {
+                if (d.errorMessage !== "") {
+                  return d.errorMessage
+                }
+                if (d.isPrivateKey) {
+                    return qsTr("Private key detected")
+                }
+                if (d.isPublicKey) {
+                    return qsTr("Public key detected")
+                }
             }
+            color: (d.isPrivateKey || d.isPublicKey) && d.errorMessage === "" ? Theme.palette.successColor1 : Theme.palette.dangerColor1
         }
     }
 
-    rightButtons: [
-        StatusQControls.StatusButton {
-            id: importButton
-            enabled: false
-            text: qsTr("Import")
-            onClicked: {
-                let communityKey = keyInput.text.trim();
-                if (!communityKey.startsWith("0x")) {
-                    communityKey = "0x" + communityKey;
-                }
 
-                root.store.importCommunity(communityKey);
-                root.close();
-            }
-        }
-    ]
+    Connections {
+      target: root.store
+      function onImportingCommunityStateChanged(communityId, state, errorMsg) {
+          let communityKey = keyInput.text.trim();
+          if (d.isPublicKey) {
+              let currentCommunityKey = Utils.isCompressedPubKey(communityKey) ?
+                  Utils.changeCommunityKeyCompression(communityKey) :
+                  communityKey
+
+              if (communityId == currentCommunityKey) {
+                  importButton.loading = false
+                  if (state === Constants.communityImported && root.opened) {
+                    root.close()
+                    return
+                  }
+              }
+
+              if (state === Constants.communityImportingError) {
+                d.errorMessage = errorMsg
```

Comment:
Ouch, the binding `property string errorMessage: isInputValid ? "" : qsTr("Invalid key")` will get broken here. Most likely when an import fails the key validation will no longer work. Mentioned issue could be reproduced in the given scenario:
- paste key
- make import fail
- paste invalid key

To fix that I would do the following:
```
QtObject {
    id: d
    property string importErrorMessage
    readonly property string inputErrorMessage: isInputValid ? "" : qsTr("Invalid key")
    readonly property string errorMessage: importErrorMessage || inputErrorMessage
    ...
}
...
onTextChanged: d.importErrorMessage = ""
```

Reply:
Will see if that fixes it but codewise I have to say this:

`onTextChanged: d.importErrorMessage = ""`

Looks very unexpected to me. "every time the text changes we want to reset the error message" 
^ This is not actually what we want (even if that is working, it's not very obvious)

Reply:
Well.. depends on what should happen when you provide a new key if the import failed before? Should previous import fail message be more important than input validation? Since you use the same component for displaying both input validation and import errors how would you indicate to the user that the new key is wrong and still show him the last import error message? To me, logic is: "whenever you provide a new key clear the last import error message" and that is well reflected by `onTextChanged: d.importErrorMessage = ""`.

Maybe we should keep both errors separate on the UI then?